src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt

```diff
@@ -77,12 +77,14 @@ class DocumentCodeTest : CodyIntegrationTextFixture() {
             }
           }
         }
-
-    FixupService.getInstance(project).addListener(showWorkingGroupSessionStateListener)
-    runAndWaitForNotifications(DocumentCodeAction.ID, TOPIC_DISPLAY_WORKING_GROUP)
-    FixupService.getInstance(project).removeListener(showWorkingGroupSessionStateListener)
-    TestCase.assertEquals(
-        "Asserts not executed or executed more than once", 1, assertsExecuted.get())
+    try {
+      FixupService.getInstance(project).addListener(showWorkingGroupSessionStateListener)
+      runAndWaitForNotifications(DocumentCodeAction.ID, TOPIC_DISPLAY_WORKING_GROUP)
+      TestCase.assertEquals(
+          "Asserts not executed or executed more than once", 1, assertsExecuted.get())
+    } finally {
+      FixupService.getInstance(project).removeListener(showWorkingGroupSessionStateListener)
+    }
   }
```

Comment:
My bad. I introduced this listener - that improved the success rate of this method but it's still not a bit flaky. The error can happen before the listener got removed (in before version). Let's remove the listener in the finally block is we can be sure it will not affect the other test cases. Other cases are now 100% green all the time.